tests/e2e/utils/helpers.ts

```diff
@@ -0,0 +1,65 @@
+import axios from 'axios'
+import config from '../config'
+import _path from 'path'
+import * as fs from 'fs'
+
+const uploadedFiles: string[] = []
+
+const getWebDavFilePath = (user, filename) => {
+  return _path.join('remote.php/dav/files', user, filename)
+}
+
+const getWebDavTrashbinPath = (user) => {
+  return _path.join('remote.php/dav/trash-bin', user)
+}
+
+const makeApiRequest = ({ method, path, data = null }): Promise<any> => {
+  try {
+    const headers = {
+      Authorization: `Basic ${Buffer.from(`${config.adminUser}:${config.adminPassword}`).toString('base64')}`
+    }
+    return axios({
+      method,
+      url: _path.join(config.baseUrlOcis, path),
+      headers,
+      data
+    })
+  } catch (error) {
+    throw new Error(error.message)
+  }
+}
+
+export const uploadFile = async (filename: string): Promise<void> => {
+  const fileUploadUrl = getWebDavFilePath(config.adminUser, filename)
+  const filePath = _path.join(config.assets, filename)
+  const fileContent = fs.readFileSync(filePath)
+  const response = await makeApiRequest({
+    method: 'PUT',
+    path: fileUploadUrl,
+    data: fileContent
+  })
+  if (response.status !== 201) {
+    throw new Error(`Failed to upload file ${filename}`)
+  }
+  uploadedFiles.push(filename)
+}
+
+const deleteFile = async (filename): Promise<void> => {
+  return await makeApiRequest({
+    method: 'DELETE',
+    path: getWebDavFilePath(config.adminUser, filename)
+  })
```

Comment:
To my understanding `pop` in line 56 is deleting the item

Reply:
why don't you checked http status code here and pop should be here IMO

Reply:
In end-of-test cleanup code, it is harder to know exactly what to do if something goes wrong. But it is nice to know that something went wrong, because the next test scenario might have trouble if stuff is left over from the previous test scenario.

I'm not sure what is happening in other UI test suites like this. My suggestion is that the code could remember any DELETEs that got a bad HTTP status, but continue to try to delete all the files. If any deletes failed, then fail the test with a message about what failed in the cleanup. That cleans up as much as possible - which at least gives the next scenario a "best chance" environment.

Reply:
status code check has been added